packages/utils/src/lib/reports/formatting.ts

```diff
@@ -58,3 +66,58 @@ export function metaDescription(
   }
   return '';
 }
+
+/**
+ * Link to local source for IDE
+ * @param source
+ * @param reportLocation
+ *
+ * @example
+ * linkToLocalSourceInIde({ file: '/src/index.ts'}, {outputDir: '/.code-pushup'}) // [/src/index.ts](../src/index.ts)
+ */
+export function linkToLocalSourceForIde(
+  source: SourceFileLocation,
+  options?: Pick<MdReportOptions, 'outputDir'>,
+): InlineText {
+  const { file, position } = source;
+  const { outputDir } = options ?? {};
+  const unixPath = toUnixPath(file);
+
+  // NOT linkable
+  if (!outputDir) {
+    return md.code(unixPath);
+  }
+
+  const relativePath = pathPosix.relative(outputDir, unixPath);
+  return md.link(formatFilePosition(relativePath, position), md.code(unixPath));
+}
+
+export function formatSourceLine(source: SourceFileLocation): string {
+  if (!source.position) {
+    return '';
+  }
+  const { startLine, endLine } = source.position;
+  return endLine
+    ? startLine === endLine
+      ? `${startLine}`
+      : `${startLine}-${endLine}`
+    : `${startLine}`;
+}
+
+export function formatFilePosition(
+  file: string,
+  position?: SourceFileLocation['position'],
+): string {
+  if (!position) {
+    return file;
+  }
+  const ide = ideEnvironment();
+  const { startLine, endLine } = position;
+  return ide === 'vscode'
+    ? `${file}#L${startLine}`
```

Comment:
Tested by running Code PushUp CLI in VSCode terminal - the links are created in Markdown and open the correct file and line :tada: